src/ipc/ipc4/helper.c

```diff
@@ -685,11 +691,20 @@ const struct comp_driver *ipc4_get_drv(uint8_t *uuid)
 
 const struct comp_driver *ipc4_get_comp_drv(int module_id)
 {
-	struct sof_man_fw_desc *desc = (struct sof_man_fw_desc *)IMR_BOOT_LDR_MANIFEST_BASE;
+	struct sof_man_fw_desc *desc = NULL;
```

Comment:
superfluous initialisation

Reply:
No, it's to handle the case below where it may or may not be initialized to IMR depending on RIMAGE_MANIFEST.

Reply:
Yes, if `RIMAGE_MANIFEST` is defined, then `desc` is assigned immediately and there's no need for initialisation. If it isn't defined, then the function returns immediately and `desc` is never used, so there's again no need to initialise. BTW, if it isn't defined and there's a `return` statement right at the beginning, then compilers / analysers might complain about a whole bunch of unused variables and unreachable code.